### PR TITLE
SOAR-5962: HTTP Unit Tests

### PR DESCRIPTION
**Pull request by pmara-r7**
## Proposed Changes

Adding unit tests to the rest plugin. 

### Description

Describe the proposed changes:

  - Unit tests for the functions within the RestApi class
  - Unit tests for actions: GET, POST, DELETE
  - Tests will not use credentials and instead use a mocked requests.request method

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [x] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [x] Screenshot of job output with the plugin changes
- [x] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [x] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [x] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [x] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [x] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [x] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [x] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [x] Work fully completed
- [ ] Functional
  - [x] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [x] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [x] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

**Review by jmcadams-r7 — DISMISSED** [review]

**Comment by jmcadams-r7** [comment]
I looked at the validation errors and he's getting the regen error even though we regenerated the plugin. Eh...approving anyway.

**Review by mberezin-r7 — DISMISSED** [review]
LGTM. Congrats! 🎂

File: plugins/rest/unit_test/mockconnection.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+class MockResponse:
+    def __init__(self, json_data, status_code, text):
+        self.json_data = json_data
+        self.status_code = status_code
+        self.text = text
+        self.headers = {}
+
+    def json(self):
+        if self.status_code == 418:
```
Comment (mrinehart-r7) [inline plugins/rest/unit_test/mockconnection.py:57]: Nice easter egg 😆

**Comment by pmara-r7** [comment]
Also another note: There are also make validate issues when I run that command locally on my machine. Not sure if its my machine or something else. Doesn't like some of the Output formatting json? That's all I got...

**Comment by jmcadams-r7** [comment]
Congrats on your first PR.

File: plugins/rest/.CHECKSUM [hunk 2]
```diff
@@ -26,6 +26,42 @@
...
+			"identifier": "hello_trigger/schema.py",
+			"hash": "6fda62f9b28a899e2ecb7aab1311bbb2"
+		},
+		{
+			"identifier": "return_bad_json_trigger/schema.py",
+			"hash": "9a207816ba5b0ab0f9e4f40d86c4cc3e"
+		},
+		{
+			"identifier": "throw_exception_trigger/schema.py",
+			"hash": "23e4aee31e1c532d9ed50830efad1b8d"
```
Comment (mrinehart-r7) [inline plugins/rest/.CHECKSUM:64]: These should not be generating into the checksum file

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]
test_utils.py has a lot of unresolved references to missing imports

File: plugins/rest/.CHECKSUM (hunk 2 above)
Comment (pmara-r7) [reply plugins/rest/.CHECKSUM:64]: This happened when I regenerated the plugin (icon-plugin regenerate)
I could manually remove these from the checksum file but I'm worried there is some other problem that will keep occurring upon regeneration

**Review by mrinehart-r7 — APPROVED** [review]

**Review by jmcadams-r7 — APPROVED** [review]

**Pull request merged**